core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -141,17 +141,10 @@ public void addNode(final @NonNull Node node) throws IOException {
 
         Node oldNode = nodes.get(node.getNodeName());
         if (node != oldNode) {
-            // TODO we should not need to lock the queue for adding nodes but until we have a way to update the
-            // computer list for just the new node
             AtomicReference<Node> old = new AtomicReference<>();
```

Comment:
With the inner class gone, there is no more need for an `AtomicReference` so this could be simplified.